modules/portal/portal-impl/src/main/java/com/enonic/xp/portal/impl/api/ApiDescriptorKeyLocator.java

```diff
@@ -0,0 +1,36 @@
+package com.enonic.xp.portal.impl.api;
+
+import java.util.Set;
+
+import com.google.common.collect.ImmutableSet;
+
+import com.enonic.xp.app.ApplicationKey;
+import com.enonic.xp.page.DescriptorKey;
+import com.enonic.xp.resource.ResourceService;
+
+public class ApiDescriptorKeyLocator
```

Comment:
why DescriptorKeyLocator is not sufficient?

Reply:
We must support both, for example, `apis/api.xml` and `apis/myapi/myapi.xml`.